Lib/tempfile.py

```diff
@@ -100,6 +100,8 @@ def _infer_return_type(*args):
     for arg in args:
         if arg is None:
             continue
+        elif isinstance(arg, _os.PathLike):
```

Comment:
Should we allow path-like objects for prefix and suffix? it will failed when path-like objects concat with string:
`file = _os.path.join(dir, pre + name + suf)`